builtin/providers/aws/resource_aws_autoscaling_attachment.go

```diff
@@ -0,0 +1,110 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsAutoscalingAttachment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsAutoscalingAttachmentCreate,
+		Read:   resourceAwsAutoscalingAttachmentRead,
+		Delete: resourceAwsAutoscalingAttachmentDelete,
+
+		Schema: map[string]*schema.Schema{
+			"group_name": &schema.Schema{
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+
+			"elb": &schema.Schema{
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAwsAutoscalingAttachmentCreate(d *schema.ResourceData, meta interface{}) error {
+	asgconn := meta.(*AWSClient).autoscalingconn
+	asgName := d.Get("group_name").(string)
+
+	elbName := d.Get("elb").(string)
+
+	attachElbInput := &autoscaling.AttachLoadBalancersInput{
+		AutoScalingGroupName: aws.String(asgName),
+		LoadBalancerNames:    []*string{aws.String(elbName)},
+	}
+
+	log.Printf("[INFO] registering asg %s with ELBs %s", asgName, elbName)
+
+	_, err := asgconn.AttachLoadBalancers(attachElbInput)
+	if err != nil {
+		return fmt.Errorf("Failure registering asg with ELBs: %s", err)
+	}
+
+	d.SetId(resource.PrefixedUniqueId(fmt.Sprintf("%s-", asgName)))
+
+	return nil
```

Comment:
Often, as a customary thing, we would invoke the `ReadFunc` so for example it would be:

```go
return resourceAwsAutoscalingAttachmentRead(d, meta)
```
This is so that it would re-read the resource back ensuring that AWS already created it, etc. Primarily as a precaution (or guard, if you wish) against the eventually consistent nature of the AWS.